src/bin/ipa_bench/config.rs

```diff
@@ -0,0 +1,134 @@
+use log::error;
+use rand_distr::num_traits::ToPrimitive;
+use serde_json::Value;
+use std::io::{BufReader, Read};
+use std::ops::Range;
+use std::process;
+
+pub fn parse<R: Read>(input: &mut R) -> Config {
+    let f = BufReader::new(input);
+
+    let config: Value = serde_json::from_reader(f).unwrap();
+
+    Config { config }
+}
+
+#[derive(Clone)]
+pub struct Config {
+    config: Value,
+}
+
+impl Config {
```

Comment:
typically for config, I like to extract all the values on init, so that I don't run into a "config missing" error half-way through executing the program. maybe not that big a deal in this case, but I've found that making sure your config is valid up front is generally helpful in the long run